src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts

```diff
@@ -9,17 +10,25 @@ import VideoUplinkBandwidthPolicy from './VideoUplinkBandwidthPolicy';
 /** NScaleVideoUplinkBandwidthPolicy implements capture and encode
  *  parameters that are nearly equivalent to those chosen by the
  *  traditional native clients, except for a modification to
- *  maxBandwidthKbps described below. */
+ *  maxBandwidthKbps and scaleResolutionDownBy described below. */
 export default class NScaleVideoUplinkBandwidthPolicy implements VideoUplinkBandwidthPolicy {
+  static readonly rid = 'video';
+
   private numParticipants: number = 0;
   private optimalParameters: DefaultVideoAndEncodeParameter;
   private parametersInEffect: DefaultVideoAndEncodeParameter;
   private idealMaxBandwidthKbps = 1400;
   private hasBandwidthPriority: boolean = false;
+  private encodingParamMap = new Map<string, RTCRtpEncodingParameters>();
+  private transceiverController: TransceiverController;
 
   constructor(private selfAttendeeId: string) {
```

Comment:
You probably need a constructor argument here so that customers can turn the resolution adjustment off or on.